[dotnet build] regression: latest cli fails to build xproj->csproj saying that TFMs are incompatible

## Steps to reproduce

1. create a new core console app
2. remove default framework and package dependencies and substitute it with "net46"
3. Create new windows classlibrary with target 4.5.1
4. Wait for restore (it is workiing fine and restores csproj ok)
5. Build

## Expected  behavior

Build fine

## Actual behavior
Build error:  Compiling ConsoleApp19 for .NETFramework,Version=v4.6
2>c:\Users\antonpis\Documents\Visual Studio 2015\Projects\ConsoleApp19\src\ConsoleApp19\project.json(13,27): error NU1002: The dependency ClassLibrary1 1.0.0 does not support framework .NETFramework,Version=v4.6.

It used to work until i installed latest dotnet cli

## Environment data
`dotnet --info` output:

C:\Windows\system32>dotnet --info
.NET Command Line Tools (1.0.0-rc2-002520)

Product Information:
 Version:     1.0.0-rc2-002520
 Commit Sha:  7c504b2456

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

--- a/src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs
+++ b/src/Microsoft.DotNet.ProjectModel/BuildWorkspace.cs
@@ -9,7 +9,7 @@ namespace Microsoft.DotNet.ProjectModel
 {
     public class BuildWorkspace : Workspace
     {
-        public BuildWorkspace(ProjectReaderSettings settings) : base(settings) { }
+        public BuildWorkspace(ProjectReaderSettings settings) : base(settings, false) { }
 
         /// <summary>
         /// Create an empty <see cref="WorkspaceContext" /> using the default <see cref="ProjectReaderSettings" />
@@ -58,7 +58,6 @@ namespace Microsoft.DotNet.ProjectModel
                     return CreateBaseProjectBuilder(context.ProjectFile)
                         .WithTargetFramework(context.TargetFramework)
                         .WithRuntimeIdentifiers(runtimeIdentifiers)
-                        .AsDesignTime()
                         .Build();
                 }
 

--- a/src/Microsoft.DotNet.ProjectModel/DesignTimeWorkspace.cs
+++ b/src/Microsoft.DotNet.ProjectModel/DesignTimeWorkspace.cs
@@ -15,7 +15,7 @@ namespace Microsoft.DotNet.ProjectModel
 
         private bool _needRefresh;
 
-        public DesignTimeWorkspace(ProjectReaderSettings settings) : base(settings) { }
+        public DesignTimeWorkspace(ProjectReaderSettings settings) : base(settings, true) { }
 
         public void AddProject(string path)
         {

--- a/src/Microsoft.DotNet.ProjectModel/Workspace.cs
+++ b/src/Microsoft.DotNet.ProjectModel/Workspace.cs
@@ -31,11 +31,13 @@ namespace Microsoft.DotNet.ProjectModel
 
         private readonly ProjectReaderSettings _settings;
         private readonly LockFileReader _lockFileReader;
+        private readonly bool _designTime;
 
-        protected Workspace(ProjectReaderSettings settings)
+        protected Workspace(ProjectReaderSettings settings, bool designTime)
         {
             _settings = settings;
             _lockFileReader = new LockFileReader();
+            _designTime = designTime;
         }
 
         public ProjectContext GetProjectContext(string projectPath, NuGetFramework framework)
@@ -168,7 +170,7 @@ namespace Microsoft.DotNet.ProjectModel
                     {
                         try
                         {
-                            currentEntry.Model = _lockFileReader.ReadLockFile(currentEntry.FilePath, fs, designTime: true);
+                            currentEntry.Model = _lockFileReader.ReadLockFile(currentEntry.FilePath, fs, designTime: _designTime);
                             currentEntry.UpdateLastWriteTimeUtc();
                         }
                         catch (FileFormatException ex)

